src/Microsoft.ML.Data/Transforms/TransformInputBase.cs

```diff
@@ -0,0 +1,25 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Data.DataView;
+using Microsoft.ML.CommandLine;
+using Microsoft.ML.EntryPoints;
+
+namespace Microsoft.ML.Transforms
+{
+    /// <summary>
+    /// The base class for all transform inputs.
+    /// </summary>
+    [TlcModule.EntryPointKind(typeof(CommonInputs.ITransformInput))]
+    public abstract class TransformInputBase
```

Comment:
>TransformInputBase [](start = 26, length = 18)

This class should have a `[BestFriend] private protected` constructor. We do not want people forming base classes willy nilly.